hadoop-yarn-project/hadoop-yarn/hadoop-yarn-api/src/main/java/org/apache/hadoop/yarn/conf/YarnConfiguration.java

```diff
@@ -4390,6 +4390,31 @@ public static boolean isAclEnabled(Configuration conf) {
   public static final String GPG_KERBEROS_PRINCIPAL_HOSTNAME_KEY = FEDERATION_GPG_PREFIX +
       "kerberos.principal.hostname";
 
+  // The application cleaner class to use
+  public static final String GPG_APPCLEANER_CLASS =
+      FEDERATION_GPG_PREFIX + "application.cleaner.class";
+  public static final String DEFAULT_GPG_APPCLEANER_CLASS =
+      "org.apache.hadoop.yarn.server.globalpolicygenerator"
```

Comment:
Can we use DefaultApplicationCleaner.getClass()?

Reply:
Thank you for the suggestion! We may not be able to make this modification because the `YarnConfiguration` is in the `hadoop-yarn-api`, and` DefaultApplicationCleaner` is in the `hadoop-yarn-server-globalpolicygenerator`. The `hadoop-yarn-server-globalpolicygenerator` indirectly depends on `hadoop-yarn-api`, and we cannot add a dependency from `hadoop-yarn-api` to `hadoop-yarn-server-globalpolicygenerator`.